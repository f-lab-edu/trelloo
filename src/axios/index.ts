import axios, { AxiosInstance } from "axios";
import { handleInterceptConfig, handleRquestInterceptError } from "./requestIntercepters";
import { handleInterceptResponse, handleResponseInterceptError } from "./responseIntercepters";

const axiosInstance = axios.create({
  headers: {
    common: {
      "Content-Type": "application/json",
    },
  },
});

const setupInterceptors = (instance: AxiosInstance): AxiosInstance => {
  instance.interceptors.request.use(handleInterceptConfig, handleRquestInterceptError);
  instance.interceptors.response.use(handleInterceptResponse, handleResponseInterceptError);

  return instance;
};

setupInterceptors(axiosInstance);

export default axiosInstance;
