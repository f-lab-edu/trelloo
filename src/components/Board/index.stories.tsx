import React from "react";
import Board from "@components/Board";

export default {
  title: "Components/Board",
  component: Board,
};

const Template = () => {
  return <Board searchKeyword="s" />;
};

export const Default = Template.bind({});
