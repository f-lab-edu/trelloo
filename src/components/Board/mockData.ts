export const mockedCardLists = {
  code: 1,
  data: [
    {
      id: "list1",
      title: "list1",
      cards: [
        {
          id: "card1",
          text: "card1",
        },
        {
          id: "card2",
          text: "card2",
        },
      ],
    },
    {
      id: "list2",
      title: "list2",
      cards: [
        {
          id: "card2-1",
          text: "card2-1",
        },
      ],
    },
  ],
};
