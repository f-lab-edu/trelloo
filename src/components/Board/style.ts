import styled from "styled-components";

interface ContainerProps {
  background: string;
}

export const Container = styled.div<ContainerProps>`
  padding: 10px 15px;
  display: flex;
  background-image: url(${({ background }) => background});
`;
