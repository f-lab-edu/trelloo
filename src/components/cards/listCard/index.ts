import AddButton from "./AddButton";
import ColorThumbnail from "./ColorThumbnail";
import ImageCard from "./ImageCard";
import ImageList from "./ImageList";
import ImageThumbnail from "./ImageThumbnail";
import Title from "./Title";

function ListCard() {}

ListCard.ImageCard = ImageCard;
ListCard.ImageThumbnail = ImageThumbnail;
ListCard.ColorThumbnail = ColorThumbnail;
ListCard.Title = Title;
ListCard.AddButton = AddButton;
ListCard.ImageList = ImageList;

export default ListCard;
