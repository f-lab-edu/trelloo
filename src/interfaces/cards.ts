export interface Card {
  id: string;
  text: string;
}

export interface CardList {
  id: string;
  title: string;
  cards: Card[];
}
