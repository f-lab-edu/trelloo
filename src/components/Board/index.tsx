import CardList from "../CardList";
import * as S from "./style";

interface Board {
  boardName: string;
}

const Board = ({ boardName }: Board) => {
  return (
    <S.Container>
      Board: {boardName}
      <CardList />
    </S.Container>
  );
};

export default Board;
