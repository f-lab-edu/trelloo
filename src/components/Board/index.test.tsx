/* eslint-disable @typescript-eslint/consistent-type-assertions */
import React from "react";
import { customRender, waitFor } from "@utils/testUtils";
import { describe, vi, expect } from "vitest";
import { mockedCardLists } from "./mockData";
import * as queries from "@/queries/cards";
import {
  AddCardRequest,
  AddListRequest,
  DeleteCardRequest,
  DeleteListRequest,
  EditCardMutationData,
  EditCardPositionRequest,
  EditCardRequest,
  EditListRequest,
  ResponseMessage,
} from "@/queries/cards/interface";
import { UseMutationResult } from "@tanstack/react-query";
import { AxiosError } from "axios";
import Board from ".";

vi.mock("@/queries/cards");

describe("Board 테스트", () => {
  const useCardsQueryMock = vi
    .spyOn(queries, "useCardsQuery")
    .mockImplementation(vi.fn().mockReturnValue({ data: mockedCardLists, isLoading: false, isSuccess: true }));

  const useAddListMutationMock = vi.fn().mockResolvedValue({ mutate: vi.fn(), message: "succeed" });
  const useEditListMutationMock = vi.fn().mockResolvedValue({ mutate: vi.fn(), message: "succeed" });
  const useDeleteListMutationMock = vi.fn().mockResolvedValue({ mutate: vi.fn(), message: "succeed" });
  const useEditCardPositionMutationMock = vi.fn().mockResolvedValue({ mutate: vi.fn(), message: "succeed" });

  const useAddCardMutationMock = vi.fn().mockResolvedValue({ mutate: vi.fn(), message: "succeed" });
  const useDeleteCardMutationMock = vi.fn().mockResolvedValue({ mutate: vi.fn(), message: "succeed" });
  const useEditCardMutationMock = vi.fn().mockResolvedValue({ mutate: vi.fn(), message: "succeed" });

  vi.spyOn(queries, "useAddListMutation").mockReturnValue(
    useAddListMutationMock as unknown as UseMutationResult<ResponseMessage, unknown, AddListRequest, unknown>,
  );
  vi.spyOn(queries, "useEditListMutation").mockReturnValue(
    useEditListMutationMock as unknown as UseMutationResult<ResponseMessage, unknown, EditListRequest, unknown>,
  );
  vi.spyOn(queries, "useDeleteListMutation").mockReturnValue(
    useDeleteListMutationMock as unknown as UseMutationResult<ResponseMessage, unknown, DeleteListRequest, unknown>,
  );
  vi.spyOn(queries, "useEditCardPositionMutation").mockReturnValue(
    useEditCardPositionMutationMock as unknown as UseMutationResult<
      ResponseMessage,
      AxiosError<unknown, any>,
      EditCardPositionRequest,
      EditCardMutationData
    >,
  );

  vi.spyOn(queries, "useAddCardMutation").mockReturnValue(
    useAddCardMutationMock as unknown as UseMutationResult<ResponseMessage, unknown, AddCardRequest, unknown>,
  );
  vi.spyOn(queries, "useEditCardMutation").mockReturnValue(
    useEditCardMutationMock as unknown as UseMutationResult<ResponseMessage, unknown, EditCardRequest, unknown>,
  );
  vi.spyOn(queries, "useDeleteCardMutation").mockReturnValue(
    useDeleteCardMutationMock as unknown as UseMutationResult<ResponseMessage, unknown, DeleteCardRequest, unknown>,
  );

  it("Board 컴포넌트 렌더링 후 useCardsQuery 호출", () => {
    customRender(<Board searchKeyword="" />);

    waitFor(() => {
      expect(useCardsQueryMock).toHaveBeenCalled();
    });
  });

  it("스냅샷 테스트", () => {
    const component = customRender(<Board searchKeyword="" />);
    expect(component).toMatchSnapshot();
  });
});
