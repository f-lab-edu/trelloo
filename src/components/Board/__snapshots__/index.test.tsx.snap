// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Board 테스트 > 스냅샷 테스트 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rbd-announcement-0"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div>
      <div
        class="sc-lnAgIa cOtAoW"
      >
        <div
          data-rbd-droppable-context-id="1"
          data-rbd-droppable-id="list1"
        >
          <div
            class="sc-dnwKUv iSzQHa"
          >
            <div
              class="ant-card sc-kgKVFY css-dev-only-do-not-override-ed5zg0"
            >
              <div
                class="ant-card-head"
                style="min-height: 40px; background-color: rgb(234, 236, 240); border-radius: 12px 12px 0 0; font-size: 14px;"
              >
                <div
                  class="ant-card-head-wrapper"
                >
                  <div
                    class="ant-card-extra"
                  >
                    <span
                      aria-label="ellipsis"
                      class="anticon anticon-ellipsis ant-dropdown-trigger"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="ellipsis"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="ant-card-body"
                style="padding: 5px 10px 10px 10px; border-radius: 0 0  12px 12px; background-color: rgb(234, 236, 240);"
              >
                <form
                  class="sc-fLQRDB iOYYdO"
                >
                  <h2
                    class="sc-bALXmG hXMxMK"
                  >
                    list1
                  </h2>
                </form>
                <div
                  aria-describedby="rbd-hidden-text-1-hidden-text-6"
                  data-rbd-drag-handle-context-id="1"
                  data-rbd-drag-handle-draggable-id="card1"
                  data-rbd-draggable-context-id="1"
                  data-rbd-draggable-id="card1"
                  draggable="false"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="sc-ktEKTO ifMYQU"
                  >
                    <div
                      class="ant-card ant-card-bordered sc-hAtEyd fnjhgi css-dev-only-do-not-override-ed5zg0"
                    >
                      <div
                        class="ant-card-body"
                        style="padding: 0px;"
                      >
                        <p>
                          card1
                        </p>
                        <span
                          aria-label="edit"
                          class="anticon anticon-edit sc-fmSAUk dEBEnP"
                          role="img"
                          tabindex="-1"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="edit"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                            />
                          </svg>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-describedby="rbd-hidden-text-1-hidden-text-6"
                  data-rbd-drag-handle-context-id="1"
                  data-rbd-drag-handle-draggable-id="card2"
                  data-rbd-draggable-context-id="1"
                  data-rbd-draggable-id="card2"
                  draggable="false"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="sc-ktEKTO ifMYQU"
                  >
                    <div
                      class="ant-card ant-card-bordered sc-hAtEyd fnjhgi css-dev-only-do-not-override-ed5zg0"
                    >
                      <div
                        class="ant-card-body"
                        style="padding: 0px;"
                      >
                        <p>
                          card2
                        </p>
                        <span
                          aria-label="edit"
                          class="anticon anticon-edit sc-fmSAUk dEBEnP"
                          role="img"
                          tabindex="-1"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="edit"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                            />
                          </svg>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
                <form>
                  <div
                    class="sc-fGFwAa iYYrfD"
                  >
                    <button
                      class="sc-gueYoa dDGBCu"
                      type="button"
                    >
                      <div
                        class="sc-dmqHEX cFVWKL"
                      >
                        <span
                          aria-label="plus"
                          class="anticon anticon-plus"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="plus"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <defs>
                              <style />
                            </defs>
                            <path
                              d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                            />
                            <path
                              d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                            />
                          </svg>
                        </span>
                      </div>
                      Add a card
                      <div
                        class="sc-hLseeU bgMlIj"
                      />
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
        <div
          data-rbd-droppable-context-id="1"
          data-rbd-droppable-id="list2"
        >
          <div
            class="sc-dnwKUv iSzQHa"
          >
            <div
              class="ant-card sc-kgKVFY css-dev-only-do-not-override-ed5zg0"
            >
              <div
                class="ant-card-head"
                style="min-height: 40px; background-color: rgb(234, 236, 240); border-radius: 12px 12px 0 0; font-size: 14px;"
              >
                <div
                  class="ant-card-head-wrapper"
                >
                  <div
                    class="ant-card-extra"
                  >
                    <span
                      aria-label="ellipsis"
                      class="anticon anticon-ellipsis ant-dropdown-trigger"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="ellipsis"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="ant-card-body"
                style="padding: 5px 10px 10px 10px; border-radius: 0 0  12px 12px; background-color: rgb(234, 236, 240);"
              >
                <form
                  class="sc-fLQRDB iOYYdO"
                >
                  <h2
                    class="sc-bALXmG hXMxMK"
                  >
                    list2
                  </h2>
                </form>
                <div
                  aria-describedby="rbd-hidden-text-1-hidden-text-6"
                  data-rbd-drag-handle-context-id="1"
                  data-rbd-drag-handle-draggable-id="card2-1"
                  data-rbd-draggable-context-id="1"
                  data-rbd-draggable-id="card2-1"
                  draggable="false"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="sc-ktEKTO ifMYQU"
                  >
                    <div
                      class="ant-card ant-card-bordered sc-hAtEyd fnjhgi css-dev-only-do-not-override-ed5zg0"
                    >
                      <div
                        class="ant-card-body"
                        style="padding: 0px;"
                      >
                        <p>
                          card2-1
                        </p>
                        <span
                          aria-label="edit"
                          class="anticon anticon-edit sc-fmSAUk dEBEnP"
                          role="img"
                          tabindex="-1"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="edit"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                            />
                          </svg>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
                <form>
                  <div
                    class="sc-fGFwAa iYYrfD"
                  >
                    <button
                      class="sc-gueYoa dDGBCu"
                      type="button"
                    >
                      <div
                        class="sc-dmqHEX cFVWKL"
                      >
                        <span
                          aria-label="plus"
                          class="anticon anticon-plus"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="plus"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <defs>
                              <style />
                            </defs>
                            <path
                              d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                            />
                            <path
                              d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                            />
                          </svg>
                        </span>
                      </div>
                      Add a card
                      <div
                        class="sc-hLseeU bgMlIj"
                      />
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-cyRfQX dOTKfH"
        >
          <form>
            <div
              class="sc-fGFwAa iYYrfD"
            >
              <button
                class="sc-gueYoa dDGBCu"
                type="button"
              >
                <div
                  class="sc-dmqHEX cFVWKL"
                >
                  <span
                    aria-label="plus"
                    class="anticon anticon-plus"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="plus"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <defs>
                        <style />
                      </defs>
                      <path
                        d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                      />
                      <path
                        d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                      />
                    </svg>
                  </span>
                </div>
                Add a list
                <div
                  class="sc-hLseeU bgMlIj"
                />
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rbd-announcement-1"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div
      id="rbd-hidden-text-1-hidden-text-6"
      style="display: none;"
    >
      
  Press space bar to start a drag.
  When dragging you can use the arrow keys to move the item around and escape to cancel.
  Some screen readers may require you to be in focus mode or to use your pass through key

    </div>
  </body>,
  "container": <div>
    <div
      class="sc-lnAgIa cOtAoW"
    >
      <div
        data-rbd-droppable-context-id="1"
        data-rbd-droppable-id="list1"
      >
        <div
          class="sc-dnwKUv iSzQHa"
        >
          <div
            class="ant-card sc-kgKVFY css-dev-only-do-not-override-ed5zg0"
          >
            <div
              class="ant-card-head"
              style="min-height: 40px; background-color: rgb(234, 236, 240); border-radius: 12px 12px 0 0; font-size: 14px;"
            >
              <div
                class="ant-card-head-wrapper"
              >
                <div
                  class="ant-card-extra"
                >
                  <span
                    aria-label="ellipsis"
                    class="anticon anticon-ellipsis ant-dropdown-trigger"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="ellipsis"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                      />
                    </svg>
                  </span>
                </div>
              </div>
            </div>
            <div
              class="ant-card-body"
              style="padding: 5px 10px 10px 10px; border-radius: 0 0  12px 12px; background-color: rgb(234, 236, 240);"
            >
              <form
                class="sc-fLQRDB iOYYdO"
              >
                <h2
                  class="sc-bALXmG hXMxMK"
                >
                  list1
                </h2>
              </form>
              <div
                aria-describedby="rbd-hidden-text-1-hidden-text-6"
                data-rbd-drag-handle-context-id="1"
                data-rbd-drag-handle-draggable-id="card1"
                data-rbd-draggable-context-id="1"
                data-rbd-draggable-id="card1"
                draggable="false"
                role="button"
                tabindex="0"
              >
                <div
                  class="sc-ktEKTO ifMYQU"
                >
                  <div
                    class="ant-card ant-card-bordered sc-hAtEyd fnjhgi css-dev-only-do-not-override-ed5zg0"
                  >
                    <div
                      class="ant-card-body"
                      style="padding: 0px;"
                    >
                      <p>
                        card1
                      </p>
                      <span
                        aria-label="edit"
                        class="anticon anticon-edit sc-fmSAUk dEBEnP"
                        role="img"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="edit"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                          />
                        </svg>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <div
                aria-describedby="rbd-hidden-text-1-hidden-text-6"
                data-rbd-drag-handle-context-id="1"
                data-rbd-drag-handle-draggable-id="card2"
                data-rbd-draggable-context-id="1"
                data-rbd-draggable-id="card2"
                draggable="false"
                role="button"
                tabindex="0"
              >
                <div
                  class="sc-ktEKTO ifMYQU"
                >
                  <div
                    class="ant-card ant-card-bordered sc-hAtEyd fnjhgi css-dev-only-do-not-override-ed5zg0"
                  >
                    <div
                      class="ant-card-body"
                      style="padding: 0px;"
                    >
                      <p>
                        card2
                      </p>
                      <span
                        aria-label="edit"
                        class="anticon anticon-edit sc-fmSAUk dEBEnP"
                        role="img"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="edit"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                          />
                        </svg>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <form>
                <div
                  class="sc-fGFwAa iYYrfD"
                >
                  <button
                    class="sc-gueYoa dDGBCu"
                    type="button"
                  >
                    <div
                      class="sc-dmqHEX cFVWKL"
                    >
                      <span
                        aria-label="plus"
                        class="anticon anticon-plus"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="plus"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <defs>
                            <style />
                          </defs>
                          <path
                            d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                          />
                          <path
                            d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                          />
                        </svg>
                      </span>
                    </div>
                    Add a card
                    <div
                      class="sc-hLseeU bgMlIj"
                    />
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
      <div
        data-rbd-droppable-context-id="1"
        data-rbd-droppable-id="list2"
      >
        <div
          class="sc-dnwKUv iSzQHa"
        >
          <div
            class="ant-card sc-kgKVFY css-dev-only-do-not-override-ed5zg0"
          >
            <div
              class="ant-card-head"
              style="min-height: 40px; background-color: rgb(234, 236, 240); border-radius: 12px 12px 0 0; font-size: 14px;"
            >
              <div
                class="ant-card-head-wrapper"
              >
                <div
                  class="ant-card-extra"
                >
                  <span
                    aria-label="ellipsis"
                    class="anticon anticon-ellipsis ant-dropdown-trigger"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="ellipsis"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                      />
                    </svg>
                  </span>
                </div>
              </div>
            </div>
            <div
              class="ant-card-body"
              style="padding: 5px 10px 10px 10px; border-radius: 0 0  12px 12px; background-color: rgb(234, 236, 240);"
            >
              <form
                class="sc-fLQRDB iOYYdO"
              >
                <h2
                  class="sc-bALXmG hXMxMK"
                >
                  list2
                </h2>
              </form>
              <div
                aria-describedby="rbd-hidden-text-1-hidden-text-6"
                data-rbd-drag-handle-context-id="1"
                data-rbd-drag-handle-draggable-id="card2-1"
                data-rbd-draggable-context-id="1"
                data-rbd-draggable-id="card2-1"
                draggable="false"
                role="button"
                tabindex="0"
              >
                <div
                  class="sc-ktEKTO ifMYQU"
                >
                  <div
                    class="ant-card ant-card-bordered sc-hAtEyd fnjhgi css-dev-only-do-not-override-ed5zg0"
                  >
                    <div
                      class="ant-card-body"
                      style="padding: 0px;"
                    >
                      <p>
                        card2-1
                      </p>
                      <span
                        aria-label="edit"
                        class="anticon anticon-edit sc-fmSAUk dEBEnP"
                        role="img"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="edit"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                          />
                        </svg>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <form>
                <div
                  class="sc-fGFwAa iYYrfD"
                >
                  <button
                    class="sc-gueYoa dDGBCu"
                    type="button"
                  >
                    <div
                      class="sc-dmqHEX cFVWKL"
                    >
                      <span
                        aria-label="plus"
                        class="anticon anticon-plus"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="plus"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <defs>
                            <style />
                          </defs>
                          <path
                            d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                          />
                          <path
                            d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                          />
                        </svg>
                      </span>
                    </div>
                    Add a card
                    <div
                      class="sc-hLseeU bgMlIj"
                    />
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-cyRfQX dOTKfH"
      >
        <form>
          <div
            class="sc-fGFwAa iYYrfD"
          >
            <button
              class="sc-gueYoa dDGBCu"
              type="button"
            >
              <div
                class="sc-dmqHEX cFVWKL"
              >
                <span
                  aria-label="plus"
                  class="anticon anticon-plus"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="plus"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <defs>
                      <style />
                    </defs>
                    <path
                      d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                    />
                    <path
                      d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                    />
                  </svg>
                </span>
              </div>
              Add a list
              <div
                class="sc-hLseeU bgMlIj"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
